tiny/lisp.c

```diff
@@ -501,7 +501,7 @@ void print(FILE *f, value_t v)
     value_t cd;
 
     switch (tag(v)) {
-    case TAG_NUM: fprintf(f, "%ld", numval(v)); break;
+    case TAG_NUM: fprintf(f, "%lld", numval(v)); break;
```

Comment:
This should have a platform check of some kind; `%ld` is correct on LP64 systems.